dino_catalog/dinoparser.rb

```diff
@@ -0,0 +1,114 @@
+class DinoParser
+  attr_reader :files, :dinosaurs
+
+  def initialize(files = [])
+    @files = files
+    @dinosaurs = []
+  end
+
+  def add(dino)
+    @dinosaurs << Dinosaur.new(standardize(dino))
+  end
+
+  def find_bipeds
+    @dinosaurs.select(&:is_biped?)
+  end
+
+  def find_carnivores
+    @dinosaurs.select(&:is_carnivore?)
+  end
+
+  def find_by_period(era)
+    @dinosaurs.select { |dino| dino.period.include? era }
+  end
+
+  def find_by_size!(size)
+    if size.downcase == "big"
+      @dinosaurs.select { |dino| !dino.weight_in_lbs.nil? && dino.weight_in_lbs > 2000 }
+    elsif size.downcase == "small"
+      @dinosaurs.select { |dino| !dino.weight_in_lbs.nil? && dino.weight_in_lbs <= 2000 }
```

Comment:
This logic is complex and domain specific. Thus, it should be pushed into the model layer:

```ruby
def small?
  !weight_in_lbs.nil? && weight_in_lbs <= 2000
end
```